src/model/HeatPumpPlantLoopEIRCooling_Impl.hpp

```diff
@@ -0,0 +1,185 @@
+#ifndef MODEL_HEATPUMPPLANTLOOPEIRCOOLING_IMPL_HPP
+#define MODEL_HEATPUMPPLANTLOOPEIRCOOLING_IMPL_HPP
+
+#include "ModelAPI.hpp"
+#include "WaterToWaterComponent_Impl.hpp"
+
+namespace openstudio {
+namespace model {
+
+  class HeatPumpPlantLoopEIRHeating;
+  class Curve;
+
+  namespace detail {
+
+    /** HeatPumpPlantLoopEIRCooling_Impl is a WaterToWaterComponent_Impl that is the implementation class for HeatPumpPlantLoopEIRCooling.*/
+    class MODEL_API HeatPumpPlantLoopEIRCooling_Impl : public WaterToWaterComponent_Impl
+    {
+     public:
+      /** @name Constructors and Destructors */
+      //@{
+
+      HeatPumpPlantLoopEIRCooling_Impl(const IdfObject& idfObject, Model_Impl* model, bool keepHandle);
+
+      HeatPumpPlantLoopEIRCooling_Impl(const openstudio::detail::WorkspaceObject_Impl& other, Model_Impl* model, bool keepHandle);
+
+      HeatPumpPlantLoopEIRCooling_Impl(const HeatPumpPlantLoopEIRCooling_Impl& other, Model_Impl* model, bool keepHandle);
+
+      virtual ~HeatPumpPlantLoopEIRCooling_Impl() {}
+
+      //@}
+      /** @name Virtual Methods */
+      //@{
+
+      virtual const std::vector<std::string>& outputVariableNames() const override;
+
+      virtual IddObjectType iddObjectType() const override;
+
+      virtual unsigned supplyInletPort() const override;
+
+      virtual unsigned supplyOutletPort() const override;
+
+      virtual unsigned demandInletPort() const override;
+
+      virtual unsigned demandOutletPort() const override;
+
+      /* This function will call the base class' method WaterToWaterComponent_Impl::addToNode()
+       * If this is connecting to the demand side of a loop, will set the condenserType to 'WaterSource'
+       */
+      virtual bool addToNode(Node& node) override;
+
+      /** Override to switch the condenser type to 'AirSource' **/
+      virtual bool removeFromSecondaryPlantLoop() override;
+
+      virtual void autosize() override;
+
+      virtual void applySizingValues() override;
+
+      //@}
+      /** @name Getters */
+      //@{
+
+      std::string condenserType() const;
+
+      boost::optional<HeatPumpPlantLoopEIRHeating> companionHeatingHeatPump() const;
+
+      boost::optional<double> referenceLoadSideFlowRate() const;
+
+      bool isReferenceLoadSideFlowRateAutosized() const;
+
+      boost::optional<double> referenceSourceSideFlowRate() const;
+
+      bool isReferenceSourceSideFlowRateAutosized() const;
+
+      boost::optional<double> referenceCapacity() const;
+
+      bool isReferenceCapacityAutosized() const;
+
+      double referenceCoefficientofPerformance() const;
+
+      double sizingFactor() const;
+
+      Curve capacityModifierFunctionofTemperatureCurve() const;
+
+      Curve electricInputtoOutputRatioModifierFunctionofTemperatureCurve() const;
+
+      Curve electricInputtoOutputRatioModifierFunctionofPartLoadRatioCurve() const;
+
+      //@}
+      /** @name Setters */
+      //@{
+
+      bool setCondenserType(const std::string& condenserType);
```

Comment:
side note: changed from `std::string` to `const std::string&` to avoid unnecessary copies